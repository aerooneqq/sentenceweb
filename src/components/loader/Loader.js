import React, {Component} from "react"
import "./LoaderStyles.css"

export default class Loader extends Component{ 
    constructor(props){ 
        super(props)
    }

    render(){ 
        return(
            <div className = "loaderCont">
                <div className = "ring"> 
                    <div></div>
                    <div></div>
                    <div></div>
                    <div></div>
                </div>
                <div className = "loaderMessageCont">
                    {this.props.message}   
                </div>
            </div>
        )
    }
}