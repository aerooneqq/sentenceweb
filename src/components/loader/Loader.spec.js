import React from 'react'
import { shallow } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import Loader from './Loader';

describe('Loader normal rendering', () => {
  it('usual props render', () => {
    const props = {
      innerWidth: 20,
      outterWidth: 30,
      message: "Test message"
    }

    const loader = shallow(<Loader {...props}/>);
    expect(loader.find("div.loaderMessageCont").text()).toBe("Test message");
  });

  it('null props render', () => {
    const props = {
      innerWidth: null,
      outterWidth: null,
      message: null
    }
    
    const loader = shallow(<Loader {...props}/>);

    expect(loader.find("div.loaderMessageCont").text()).toBe("");
    expect(loader).toMatchSnapshot()
  });

  it("undefined props render", () => { 
    const props = { 
      innerWidth: undefined,
      outterWidth: undefined,
      message: undefined 
    }

    const loader = shallow(<Loader {...props}/>);

    expect(loader.find("div.loaderMessageCont").text()).toBe("");
    expect(loader).toMatchSnapshot()
  });
})