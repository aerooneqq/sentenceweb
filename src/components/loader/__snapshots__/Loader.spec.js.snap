// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader normal rendering null props render 1`] = `ShallowWrapper {}`;

exports[`Loader normal rendering should render correctly 1`] = `
<div
  className="loaderCont"
>
  <div
    className="ring"
    style={
      Object {
        "height": "64px",
        "width": "64px",
      }
    }
  >
    <div
      style={
        Object {
          "height": "51px",
          "width": "51px",
        }
      }
    />
    <div
      style={
        Object {
          "height": "51px",
          "width": "51px",
        }
      }
    />
    <div
      style={
        Object {
          "height": "51px",
          "width": "51px",
        }
      }
    />
    <div
      style={
        Object {
          "height": "51px",
          "width": "51px",
        }
      }
    />
  </div>
  <div
    className="loaderMessageCont"
  />
</div>
`;

exports[`Loader normal rendering undefined props render 1`] = `ShallowWrapper {}`;
